crates/sqlexec/src/planner/physical_plan/drop_credentials.rs

```diff
@@ -20,13 +20,22 @@ use datafusion::physical_plan::{
 use futures::stream;
 use protogen::metastore::types::service::{self, Mutation};
 
-use super::{new_operation_batch, GENERIC_OPERATION_PHYSICAL_SCHEMA};
+use super::{
+    new_operation_batch,
+    new_operation_with_deprecation_batch,
+    GENERIC_OPERATION_PHYSICAL_SCHEMA,
+    GENERIC_OPERATION_WITH_DEPRECATION_PHYSICAL_SCHEMA,
+};
+
+const DEPRECATION_MESSAGE: &str = "`DROP CREDENTIALS` is deprecated and will be removed in a future release. Use `DROP CREDENTIAL` instead.";
 
 #[derive(Debug, Clone)]
 pub struct DropCredentialsExec {
     pub catalog_version: u64,
     pub names: Vec<String>,
     pub if_exists: bool,
+    /// If true, show a deprecation warning when the operation is executed.
+    pub show_deprecation_warning: bool,
```

Comment:
To my mind, the deprecation is a static/implementation property of the type, and "show deprecation warning" is a property of the session/client and not the physical plan. 

Is there a reason that I've missed that means that the display/output of the struct has to be mutable. 

-----

I've read this again, and I think my confusion is that this struct is `DropCredntials` and should be renamed to match the output. I get why this is here, but it feels like it makes it difficult to generalize, and that we'll have to do this work basically every time we want to change syntax or deprecate something.

Reply:
yeah I considered renaming it to `DropCredential`, but didn't want to overcomplicate the PR. Will update with the rename.